File: Momento_2/Bloque_6/Editorial/C++/src/Libro.cpp
#include "../include/Libro.h"

Libro::Libro() : Publicacion(), numeroPaginas(0), anioPublicacion(0) {}
Libro::Libro(const std::string &titulo, const std::string &autor, int numeroPaginas, int anioPublicacion)
    : Publicacion(titulo, autor), numeroPaginas(numeroPaginas), anioPublicacion(anioPublicacion) {}

std::string Libro::mostrar() const
{
    return "Libro =[" + Publicacion::mostrar() + ", " + std::to_string(numeroPaginas) + " paginas, " + std::to_string(anioPublicacion) + "]";
}

std::ostream &operator<<(std::ostream &os, const Libro &libro)
{
    os << libro.mostrar();
    return os;
}

File: Momento_2/Bloque_6/Editorial/JAVA/src/models/Libro.java
package models;

public class Libro extends Publicacion {
    private int numeroPaginas;
    private int anioPublicacion;

    public Libro() {}

    public Libro(String titulo, double precio, int numeroPaginas, int anioPublicacion) {
        super(titulo, precio);
        this.numeroPaginas = numeroPaginas;
        this.anioPublicacion = anioPublicacion;
    }

    public int getNumeroPaginas() {
        return numeroPaginas;
    }

    public void setNumeroPaginas(int numeroPaginas) {
        this.numeroPaginas = numeroPaginas;
    }

    public int getAnioPublicacion() {
        return anioPublicacion;
    }

    public void setAnioPublicacion(int anioPublicacion) {
        this.anioPublicacion = anioPublicacion;
    }

    @Override
    public String toString() {
        return "Libro{" + "numeroPaginas=" + numeroPaginas + ", anioPublicacion=" + anioPublicacion
                + "} " + super.toString();
    }
}
